File: leetCode/Medium/Minimum Number of Operations to Move All Balls to Each Box.ts
function minOperations(boxes: string): number[] {
    const answer: number[] = [];
    const boxList = boxes.split("");

    for (let i = 0; i < boxList.length; i++) {
        let result = 0;

        for (let j = 0; j < boxList.length; j++) {
            if (i != j && boxList[j] === "1") {
                result += Math.abs(j - i);
            }
        }

        answer.push(result);
    }

    return answer;
}

console.log(minOperations("110")); //[1,1,3]
console.log(minOperations("001011")); //[11,8,5,4,3,4]

File: leetCode/Medium/Minumum_Number_of_Operations_to_move_all_balls_to_each_box/Minimum_number_of_operations_to_move_all_balls_to_each_box.cpp
#include <cmath>
#include <iostream>
#include <vector>

using namespace std;

vector<int> minOperations(string boxes) {
  vector<int> answer;
  int n = boxes.size();

  for (int i = 0; i < n; i++) {
    int result = 0;
    for (int j = 0; j < n; j++) {
      if (i != j && boxes[j] == '1') {
        result += abs(j - i);
      }
    }
    answer.push_back(result);
  }

  return answer;
}

int main() {
  vector<int> result1 = minOperations("110");
  for (int num : result1) {
    cout << num << " ";
  }
  cout << endl;

  vector<int> result2 = minOperations("001011");
  for (int num : result2) {
    cout << num << " ";
  }
  cout << endl;

  return 0;
}
